scripts/pre_commit_linter.py

```diff
@@ -1927,6 +1934,146 @@ def _check_for_copyright_notice(self):
+                    if os.path.join(root, file_name) in CODEOWNER_FILE_PATHS:
+                        is_root_file = True
+                    if (any(root.startswith(
+                            dir_path) for dir_path in CODEOWNER_DIR_PATHS)
+                            or is_root_file):
+                        match = False
+                        # Ignore .pyc and __init__.py files.
+                        if file_name.endswith(
+                                '.pyc') or file_name == '__init__.py':
+                            match = True
+                            continue
+                        for path_to_match in path_patterns:
+                            # The level of the file in the directory
+                            # structure. For e.g. /core/controllers/
+                            # domain.py is on third level.
+                            # This condition checks if the path to check
+                            # is a directory or a file. If it is a
+                            # file, the level would be the same as found
+                            # by calculating len(path_to_match.split('/'))
+                            # but is reduced by one if it is a directory
+                            # since the split command will return an empty
+                            # string at the last of the list which will
+                            # wrongfully increase the level.
+                            if path_to_match.split('/')[-1]:
+                                level = len(path_to_match.split('/'))
+                            else:
+                                level = len(path_to_match.split('/')) - 1
+                            # This condition finally matches the file being
+                            # walked currently against the path from the
+                            # CODEOWNERS file. The level helps in matching
+                            # by considering the file name upto only the
+                            # the level of the CODEOWNERS path. For e.g.
+                            # if the file being walked upon is /core/domain
+                            # /domain.py and the path from CODEOWNERS to be
+                            # matched is /core/ then it will only consider
+                            # the file name upto level 1 i.e. just the '/core'
+                            # part of the file name since it is sufficient to
+                            # be matched.
+                            if os.path.join(*((
+                                    os.path.join(root, file_name).replace(
+                                        './', '', 1)).split(
+                                            '/')[:level])) in glob.glob(
+                                                path_to_match):
+                                match = True
+                                break
+                        if not match and self.verbose_mode_enabled:
```

Comment:
@DubeySandeep -- is our codeowner file now fully populated? I thought one reason for this check was to ensure that it is.

Reply:
Not exactly we have only covered the independent directories, but folders like core/domain/ cannot have a single co-owner so lots of file in that folder doesn't have co-owners.

I tried running `git check-ignore` on the core/domain/* and found there are 89 files in that dir which isn't assigned to anyone.

<details>
<summary>List of files (89 items)</summary>
<br><br>
<pre>
::  core/domain/action_registry.py
::  core/domain/action_registry_test.py
::  core/domain/activity_domain.py
::  core/domain/activity_domain_test.py
::  core/domain/activity_jobs_one_off.py
::  core/domain/activity_jobs_one_off_test.py
::  core/domain/activity_services.py
::  core/domain/activity_services_test.py
::  core/domain/calculation_registry.py
::  core/domain/calculation_registry_test.py
::  core/domain/config_domain.py
::  core/domain/config_domain_test.py
::  core/domain/config_services.py
::  core/domain/customization_args_util.py
::  core/domain/customization_args_util_test.py
::  core/domain/dependency_registry.py
::  core/domain/dependency_registry_test.py
::  core/domain/email_domain.py
::  core/domain/email_domain_test.py
::  core/domain/email_jobs_one_off.py
::  core/domain/email_jobs_one_off_test.py
::  core/domain/email_manager.py
::  core/domain/email_manager_test.py
::  core/domain/email_services.py
::  core/domain/email_services_test.py
::  core/domain/email_subscription_services.py
::  core/domain/email_subscription_services_test.py
::  core/domain/event_services.py
::  core/domain/event_services_test.py
::  core/domain/fs_domain.py
::  core/domain/fs_domain_test.py
::  core/domain/fs_services.py
::  core/domain/__init__.py
::  core/domain/interaction_registry.py
::  core/domain/interaction_registry_test.py
::  core/domain/learner_playlist_services.py
::  core/domain/learner_playlist_services_test.py
::  core/domain/learner_progress_domain.py
::  core/domain/learner_progress_domain_test.py
::  core/domain/learner_progress_services.py
::  core/domain/learner_progress_services_test.py
::  core/domain/moderator_services.py
::  core/domain/moderator_services_test.py
::  core/domain/obj_services.py
::  core/domain/obj_services_test.py
::  core/domain/param_domain.py
::  core/domain/param_domain_test.py
::  core/domain/playthrough_issue_registry.py
::  core/domain/playthrough_issue_registry_test.py
::  core/domain/rating_services.py
::  core/domain/rating_services_test.py
::  core/domain/recommendations_jobs_one_off.py
::  core/domain/recommendations_jobs_one_off_test.py
::  core/domain/recommendations_services.py
::  core/domain/recommendations_services_test.py
::  core/domain/rights_manager.py
::  core/domain/rights_manager_test.py
::  core/domain/role_services.py
::  core/domain/role_services_test.py
::  core/domain/rte_component_registry.py
::  core/domain/rte_component_registry_test.py
::  core/domain/search_services.py
::  core/domain/search_services_test.py
::  core/domain/state_domain.py
::  core/domain/state_domain_test.py
::  core/domain/subscription_services.py
::  core/domain/subscription_services_test.py
::  core/domain/subtopic_page_domain.py
::  core/domain/subtopic_page_domain_test.py
::  core/domain/subtopic_page_services.py
::  core/domain/subtopic_page_services_test.py
::  core/domain/summary_services.py
::  core/domain/summary_services_test.py
::  core/domain/user_domain.py
::  core/domain/user_domain_test.py
::  core/domain/user_jobs_continuous.py
::  core/domain/user_jobs_continuous_test.py
::  core/domain/user_jobs_one_off.py
::  core/domain/user_jobs_one_off_test.py
::  core/domain/user_query_jobs_one_off.py
::  core/domain/user_query_jobs_one_off_test.py
::  core/domain/user_query_services.py
::  core/domain/user_query_services_test.py
::  core/domain/user_services.py
::  core/domain/user_services_test.py
::  core/domain/value_generators_domain.py
::  core/domain/value_generators_domain_test.py
::  core/domain/visualization_registry.py
::  core/domain/visualization_registry_test.py
</pre>
</details>

Note: This is only the list of core/domain/* files which doesn't have codeowners.

Reply:
Oh, cool, I didn't know it was possible to do expandables in github comments :D 

Hm -- so, what is the plan for these 89 files (and others)? Are we going to find a way to assign them codeowners? I thought we wanted every file in the codebase to be covered.

Also, @YashJipkate, isn't this supposed to show up prominently in the Travis logs, even if it's a warning? I looked at the lint logs briefly but didn't see the list of "has no code owner" files.

Reply:
@seanlip, the check is part of lint tests, so it shows in the CircleCI logs. The Travis runs only the E2E tests.